### PR TITLE
Added a filter to skip the already stored tags and permits to only calculate hashes for the new tags

### PR DESCRIPTION
**Pull request by JulienTD**
## Description

This pull request adds a filter to skip the already stored tags and permits to only calculate hashes for the new tags

## Related Issue(s)
 
N/A

### CODE REVIEW

**Comment by polyedre** [comment]
I do not understand what this commit does : why is it necessary to remove existing tags ?

**Comment by JulienTD** [comment]
Sorry this commit is not very explicit, this PR adds a new feature that permits to retrieve first all the tags stored in the HashThePlanet's database for a technology (for instance: jquery), then it will retrieve all current tags from GitHub for instance, then it will compare them and keep only the different tags in the goal to only keep the new tags and calculates new hashes.

File: tests/resources/test_git_resource.py [hunk 1]
```diff
@@ -276,6 +294,83 @@ def session_scope():
...
+    stored_versions = [
+        MockedVersionTable("A"),
+    ]
+    repository_versions = [
+        MockedTag("A"),
+    ]
+
+    result = git_resource._filter_stored_tags(stored_versions, repository_versions)
+
+    assert list(map(lambda tag: tag.name, result)) == []
```
Comment (bretfourbe) [inline tests/resources/test_git_resource.py:371]: list comprehension ?

File: tests/resources/test_git_resource.py [hunk 2]
```diff
@@ -276,6 +294,83 @@ def session_scope():
...
+    repository_versions = [
+        MockedTag("B"),
+        MockedTag("C"),
+        MockedTag("D"),
+        MockedTag("E")
+    ]
+
+    result = git_resource._filter_stored_tags(stored_versions, repository_versions)
+
+    assert list(map(lambda tag: tag.name, result)) == ["B", "C", "D", "E"]
```
Comment (bretfourbe) [inline tests/resources/test_git_resource.py:342]: list comprehension

File: tests/resources/test_git_resource.py [hunk 3]
```diff
@@ -276,6 +294,83 @@ def session_scope():
...
+        MockedTag("A"),
+        MockedTag("B"),
+        MockedTag("C"),
+        MockedTag("D"),
+        MockedTag("E")
+    ]
+
+    result = git_resource._filter_stored_tags(stored_versions, repository_versions)
+
+    assert list(map(lambda tag: tag.name, result)) == ["D", "E"]
```
Comment (bretfourbe) [inline tests/resources/test_git_resource.py:326]: list comprehension

File: tests/resources/test_git_resource.py [hunk 4]
```diff
@@ -276,6 +294,83 @@ def session_scope():
...
+        MockedTag("A"),
+        MockedTag("B"),
+        MockedTag("C"),
+        MockedTag("D"),
+        MockedTag("E")
+    ]
+
+    result = git_resource._filter_stored_tags(stored_versions, repository_versions)
+
+    assert list(map(lambda tag: tag.name, result)) == ["B", "C", "D", "E"]
```
Comment (bretfourbe) [inline tests/resources/test_git_resource.py:360]: idem

File: tests/resources/test_git_resource.py (hunk 1 above)
Comment (bretfourbe) [reply tests/resources/test_git_resource.py:371]: assert not [...] without comparison

File: tests/resources/test_git_resource.py [hunk 5]
```diff
@@ -121,6 +122,23 @@ def test_hash_files():
...
+
+        assert sp_mock.call_count == 1
+        sp_mock.assert_called_with(['git', 'cat-file', '-p', 'empty'], shell=False)
+
+        getcwd_mock.assert_called_once()
+
+        assert chdir_mock.call_count == 2
+        chdir_mock.assert_called_with("/foobar/")
+
+        assert len(files_metadata) == 0
```
Comment (bretfourbe) [inline tests/resources/test_git_resource.py:139]: assert not files_metadata ?

File: hashtheplanet/resources/git_resource.py [hunk 6]
```diff
@@ -183,6 +184,29 @@ def _save_hashes(
...
+            return []
+        stored_tags_name = list(map(lambda element: element.version, stored_versions))
+        for found_tag_idx, found_tag in enumerate(found_tags):
+            last_found_tag_idx = found_tag_idx - 1
+
+            if found_tag_idx >= len(stored_tags_name) or found_tag.name != stored_tags_name[found_tag_idx]:
+
+                # this verification permits to know if it's the first to be added,
+                # and if it's the case, then we add the one before to permits to make a diff
+                if last_found_tag_idx >= 0 and len(result) == 0:
```
Comment (bretfourbe) [inline hashtheplanet/resources/git_resource.py:205]: and not result ?

**Pull request merged**